[dynamo 3.11] enable dynamo unittests in 3.11

williamwen42 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #96511
* __->__ #98104

Enable most dynamo unittests for 3.11. There are a few tests that are skipped due to failures that will be addressed in upcoming PRs.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov @soumith @anijain2305 @Xia-Weiwen @peterbell10

williamwen42 committed:
[dynamo 3.11] enable dynamo unittests in 3.11

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo 3.11] enable dynamo unittests in 3.11"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

albanD commented on tools/dynamo/verify_dynamo.py:
```diff
@@ -215,9 +215,8 @@ def main():
-        # TODO remove check when 3.11 is supported
-        if sys.version_info >= (3, 11):
-            warnings.warn("Dynamo not yet supported in Python 3.11. Skipping check.")
+        if sys.version_info >= (3, 12):
```
@e should leave a warning for 3.11 until everything is working fine no?

williamwen42 replied:
I'll leave a warning in eval_frame.py

williamwen42 committed:
Update on "[dynamo 3.11] enable dynamo unittests in 3.11"


Enable most dynamo unittests for 3.11. There are a few tests that are skipped due to failures that will be addressed in upcoming PRs.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

albanD reviewed (approved):
SGTM when CI is green!

williamwen42 committed:
Update on "[dynamo 3.11] enable dynamo unittests in 3.11"


Enable most dynamo unittests for 3.11. There are a few tests that are skipped due to failures that will be addressed in upcoming PRs.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo 3.11] enable dynamo unittests in 3.11"


Enable most dynamo unittests for 3.11. There are a few tests that are skipped due to failures that will be addressed in upcoming PRs.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

williamwen42 commented:
@pytorchbot merge

williamwen42 committed:
Update on "[dynamo 3.11] enable dynamo unittests in 3.11"


Enable most dynamo unittests for 3.11. There are a few tests that are skipped due to failures that will be addressed in upcoming PRs.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

williamwen42 commented:
@pytorchbot merge

malfet commented:
@pytorchbot merge -ic

huydhn commented:
@pytorchbot revert -m 'Sorry for reverting your PR, but it is failing on CPU 3.11 test in trunk https://hud.pytorch.org/pytorch/pytorch/commit/0066f3405f290ab6ef379abea6945058f8eb7ce5.  This is probably a landrace' -c landrace

huydhn commented:
@pytorchbot rebase -b master

williamwen42 committed:
Update on "[dynamo 3.11] enable dynamo unittests in 3.11"


Enable most dynamo unittests for 3.11. There are a few tests that are skipped due to failures that will be addressed in upcoming PRs.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

williamwen42 commented:
Skipping inductor minifier tests on Py 3.11+ since a bunch of the tests involve raising exceptions that aren't handled, which cause segfaults instead of a nicely printed stack trace.

williamwen42 committed:
Update on "[dynamo 3.11] enable dynamo unittests in 3.11"


Enable most dynamo unittests for 3.11. There are a few tests that are skipped due to failures that will be addressed in upcoming PRs.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

williamwen42 commented:
@pytorchbot merge
